pkg/registry/scheduling/rest/storage_scheduling.go

```diff
@@ -108,46 +114,107 @@ func (p RESTStorageProvider) v1Storage(apiResourceConfigSource serverstorage.API
 }
 
 func (p RESTStorageProvider) PostStartHook() (string, genericapiserver.PostStartHookFunc, error) {
-	return PostStartHookName, AddSystemPriorityClasses(), nil
+	return PostStartHookName, addSchedulingDefaults(), nil
 }
 
-func AddSystemPriorityClasses() genericapiserver.PostStartHookFunc {
+// addSchedulingDefaults adds the default cluster critical priorityClasses and clusterResourceQuota which allows
+// unlimited number of critical pods be created in `kube-system` namespace.
+func addSchedulingDefaults() genericapiserver.PostStartHookFunc {
 	return func(hookContext genericapiserver.PostStartHookContext) error {
 		// Adding system priority classes is important. If they fail to add, many critical system
 		// components may fail and cluster may break.
-		err := wait.Poll(1*time.Second, 30*time.Second, func() (done bool, err error) {
-			schedClientSet, err := schedulingclient.NewForConfig(hookContext.LoopbackClientConfig)
-			if err != nil {
-				utilruntime.HandleError(fmt.Errorf("unable to initialize client: %v", err))
-				return false, nil
-			}
+		if err := addSystemPriorityClasses(hookContext); err != nil {
+			return fmt.Errorf("unable to add default system priority classes: %v", err)
+		}
+		// Add default resource quota which allows unlimited number of critical pods to be created in kube-system
+		// namespace. This is to ensure backwards compatibility with existing system where we limit critical pods
+		// to be created in `kube-system`.
+		if err := addDefaultSystemQuota(hookContext); err != nil {
```

Comment:
I wouldn't have expected the scheduling storage to create resources in an unrelated API group. narrowly, this storage doesn't even know if quota is available or enabled, right?

Reply:
Good point. The main reason I added it here was we're adding `defaultQuota` to limit priorityClasses which is a scheduling construct.